components/modal/src/ModalHeader.js

```diff
@@ -0,0 +1,55 @@
+/** @jsx jsx */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme } from '@westpac/core';
+import { CloseIcon } from '@westpac/icon';
+import { Title } from './styled';
+
+// ==============================
+// Component
+// ==============================
+export const ModalHeader = ({ id, onClose, children, ...props }) => {
+	const { modal, colors } = useTheme();
+
+	return (
+		<div
+			css={{
+				display: 'flex',
+				justifyContent: 'space-between',
+				alignItems: 'center',
+				overflow: 'hidden',
+				borderBottom: `${modal.header.borderWidth} solid ${modal.header.borderColor}`,
+				padding: modal.header.padding,
+			}}
+			{...props}
+		>
+			<Title id={id}>{children}</Title>
+			<button
+				css={{
+					border: 'none',
+					padding: '5px 12px',
+					cursor: 'pointer',
+					touchAction: 'manipulation',
+					alignSelf: 'flex-start',
```

Comment:
You can remove this (_see above_). The flex parent can control the title and button children, both should align to the top really.